tests/evaluate/test_default_evaluator.py

```diff
@@ -3383,7 +3402,7 @@ def test_evaluate_precision_at_k_no_model_type():
     X = pd.DataFrame({"question": ["q1?"] * 3, "ground_truth": [("doc1", "doc2")] * 3})
 
     def fn(X):
-        return pd.DataFrame({"retrieved_context": [("doc1", "doc3", "doc2")] * len(X)})
```

Comment:
Why should this be removed? Besides, do we have any recommendation in our docs to say what data format should be returned from the model/function?

Reply:
Oh so there is already a test with pandas.dataframe: https://github.com/mlflow/mlflow/blob/7a924c78704043b4caede54bc66d9574b6cc0590/tests/evaluate/test_default_evaluator.py#L3221

I just wanted to test the dict usecase as well. I think the return type of the predict is already mentioned. We support all those types